### PR TITLE
New architecture

### PR DESCRIPTION
**Pull request by denis-mludek**
New app port : 9000

### CODE REVIEW

**Comment by xaviermichel** [comment]
New port (9000) has to be updated in Vagrantfile :)

**Comment by MythX** [comment]
On your PR you have 3 old commits.
Rebase and push.

File: server/config/socketio.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+var config = require('./environment');
+
+// When the user disconnects.. perform this
+function onDisconnect(socket) {}
+
+// When the user connects.. perform this
+function onConnect(socket) {
+    // When the client emits 'info', this listens and executes
+    socket.on('info', function(data) {
+        console.info('[%s] %s', socket.address, JSON.stringify(data, null, 2));
```
Comment (xaviermichel) [inline server/config/socketio.js]: Please use logger

File: server/config/socketio.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        socket.address = socket.handshake.address !== null ?
+            socket.handshake.address.address + ':' + socket.handshake.address.port :
+            process.env.DOMAIN;
+
+        socket.connectedAt = new Date();
+
+        // Call onDisconnect.
+        socket.on('disconnect', function() {
+            onDisconnect(socket);
+            console.info('[%s] DISCONNECTED', socket.address);
```
Comment (xaviermichel) [inline server/config/socketio.js]: logger

File: server/config/socketio.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        });
+
+        // Call onConnect.
+        onConnect(socket);
+        console.info('[%s] CONNECTED', socket.address);
```
Comment (xaviermichel) [inline server/config/socketio.js]: re-logger :)

Comment (denis-mludek) [inline server/config/socketio.js]: This is not used for the moment. It will be corrected with the next refactor :)

**Comment by denis-mludek** [comment]
Vagrantfile corrected :)

**Comment by xaviermichel** [comment]
Sounds good for me ! :+1:

File: server/views/404.html [hunk 4]
```diff
@@ -0,0 +1,157 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <title>Page Not Found :(</title>
+    <style>
```
Comment (MythX) [inline server/views/404.html]: Can you create a new css file (for example : 404.css).
I think it is better to separate the style from a CSS file.

**Pull request merged**